File: examples/languages/go-multimodule/source/geometry/geometry.go
// Package geometry is a SECOND Go package in this module, imported by main.
// It demonstrates Go's identifier-case encapsulation: capitalized identifiers
// (RectangleArea, RectanglePerimeter) are exported across the package
// boundary; the lowercase `scale` is package-private and cannot be named by
// `main`.
package geometry

// RectangleArea is exported: callable from other packages.
func RectangleArea(width, height int) int {
	return scale(width, height)
}

// RectanglePerimeter is exported.
func RectanglePerimeter(width, height int) int {
	return 2 * (width + height)
}

// scale is unexported (lowercase): internal to this package only.
func scale(a, b int) int {
	return a * b
}

File: examples/languages/rust-multimodule/source/geometry.rs
//! `geometry`: a sibling module (`source/geometry.rs`) resolved by Cargo from
//! the crate root via `mod geometry;` in `main.rs`.
//!
//! Demonstrates encapsulation: `rectangle_area` is the only `pub` item;
//! `scale` is a PRIVATE helper that callers outside this module cannot name.

/// Area of an axis-aligned rectangle. The only exported item of this module.
pub fn rectangle_area(width: i64, height: i64) -> i64 {
    // Uses the private helper, proving an internal-only seam.
    scale(width, height)
}

/// Private helper: multiply two extents. Not `pub`, so `main` (or any other
/// module) cannot call `geometry::scale` - it is encapsulated here.
fn scale(a: i64, b: i64) -> i64 {
    a * b
}
